### PR TITLE
Introduce option to control the generation of the "Do not edit" header

### PR DESCRIPTION
**Pull request by didoo**
*Issue #107*

*Description of changes:*
As discussed in #107, I have introduced a way to control the presence of a "Do not edit (plus timestamp)" header in the generated files. 

For backward compatibility, by default the header is generated, and the user has to explicitly set `showHeading=false` to disable it.

I have also updated the documentation accordingly. Let me know if there is something else to update that I may have missed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/common/formats.js [hunk 1]
```diff
@@ -34,7 +40,7 @@ function variablesWithPrefix(prefix, properties) {
 }
 
 function iconsWithPrefix(prefix, properties, config) {
-  return fileHeader() + _.chain(properties)
+  return fileHeader(config.options) + _.chain(properties)
```
Comment (dbanksdesign) [inline lib/common/formats.js]: `config` here is referencing the platform object and not the file object, like `this` refers to inside the formatter function (like on line 74 below). Should we change the 3rd argument to just be 'options' and then the formatter passes only the options object? I'm assuming we are only going to do file level options for this PR and do the cascading options in a later one.

File: lib/common/formats.js [hunk 2]
```diff
@@ -12,13 +12,19 @@
...
 
-function fileHeader() {
-  return '/**\n' +
-         ' * Do not edit directly\n' +
-         ' * Generated on ' + new Date() + '\n' +
-         ' */\n\n';
+function fileHeader(options) {
+  var to_ret = '';
+  // for backward compatibility we need to have the user explicitly hide them
+  var showHeader = (options) ? options.showHeader : true;
```
Comment (dbanksdesign) [inline lib/common/formats.js]: What are your thoughts about renaming this? I'm thinking if we eventually want options to cascade (root > platform > file), we might want this option name to be more explicit it is about files. Open to suggestions, maybe showFileComment ?

File: lib/common/formats.js [hunk 3]
```diff
@@ -12,13 +12,19 @@
  */
 
 var _ = require('lodash');
-var module_def = 'module.exports = ';
```
Comment (dbanksdesign) [inline lib/common/formats.js]: Nice, makes sense to remove this as it's only being used once.

File: lib/common/formats.js (hunk 1 above)
Comment (dbanksdesign) [reply lib/common/formats.js]: Now that I look at it, our iconsWithPrefix function is a bit wonky and confusing. Maybe for now, what if we removed the `fileHeader()` in iconsWithPrefix and moved it to the formatter functions themselves so it matches variablesWithPrefix ?

File: README.md [hunk 4]
```diff
@@ -142,6 +142,8 @@ This tells the style dictionary build system how and what to build. The default
 | platform.file.destination | String (optional) | Location to build the file, will be appended to the buildPath. |
 | platform.file.format | String (optional) | Format used to generate the file. Can be a built-in one or you can create your own. Must declare a format or a template. |
 | platform.file.template | String (optional) | Template used to generate the file. Can be a built-in one or you can create your own. [More on formats and templates](https://amzn.github.io/style-dictionary/formats_and_templates) |
+| platform.options | Object (optional) | A set of extra options associated with the file. |
```
Comment (dbanksdesign) [inline README.md]: I think this should be `platform.file.options` right?

Comment (didoo) [reply README.md]: You're right. My bad, it was a leftover of the other "variant".

File: lib/common/formats.js (hunk 1 above)
Comment (didoo) [reply lib/common/formats.js]: yes, great idea!

File: lib/common/formats.js (hunk 2 above)
Comment (didoo) [reply lib/common/formats.js]: I went for `showFileHeader`. Do you think it's clear enough?

**Comment by chazzmoney** [comment]
Switching to develop branch

**Comment by gearsandcode** [comment]
@didoo - can you edit your initial comment here to reflect the name of the option as showHeader: false vs. showHeading=false please.

**Comment by didoo** [comment]
@gearsandcode is it ok now?

**Pull request merged**